frontend/src/metabase-types/api/visualization-settings.ts

```diff
@@ -0,0 +1,16 @@
+// SmartScalar (Trend Chart)
+export type SelectedComparisonPeriodsAgo = {
+  type: "periodsAgo";
+  value: number;
+};
+type SelectedComparisonPreviousPeriod = {
+  type: "previousPeriod";
+};
+type SelectedComparisonCompareToPrevious = {
+  type: "previousValue";
+};
+
+export type SelectedComparison =
+  | SelectedComparisonCompareToPrevious
+  | SelectedComparisonPreviousPeriod
+  | SelectedComparisonPeriodsAgo;
```

Comment:
nit: I think the word "selected" makes way more sense in the UI context (when used with some input component). Should these names be a bit more generic? Maybe just `ComparisonPreviousPeriod` or `SmartScalarComparisonPreviousPeriod`